Sources/Sentry/Public/Sentry.h

```diff
@@ -42,11 +42,13 @@ FOUNDATION_EXPORT const unsigned char SentryVersionString[];
 #    import <Sentry/SentrySpanProtocol.h>
 #    import <Sentry/SentrySpanStatus.h>
 #    import <Sentry/SentryStacktrace.h>
+#    import <Sentry/SentrySwift.h>
```

Comment:
`h`: Doesn't this make all Swift classes public?

I can now access `NSArray<NSString *> * features = [SentryEnabledFeaturesBuilder getEnabledFeaturesWithOptions:options];` in the AppDelegate.m from the iOS-Objective-C sample app.

Reply:
Yes, it does for Objc project.

If you think this is a problem we close the PR, I tried other ways to solve this without success, its this or nothing.
I dont think users will start to use things they dont know how to use. It's already hard to make them use the public APIs that have documentation.

Also, this will be a big problem for some users with special projects, like this one:  #4543

Reply:
> I dont think users will start to use things they dont know how to use.

Murphy's law: "Anything that can go wrong will go wrong." So someone will use these classes. They always do. If I'm not mistaken, there is no difference between an actual public class in ObjC and an accidentally public class as `SentryEnabledFeaturesBuilder`. So, how can I tell the difference between the two in ObjC code? I didn't look closely into this, but maybe a stupid question: Can't we split up the classes into two headers? We can also discuss this on a call, if this turns out the be a longer discussion.

Reply:
Also, didn't we want to move the SR code into its own module / SPM / Cocoa package? Maybe this would help here as well.

Reply:
> Also, didn't we want to move the SR code into its own module / SPM / Cocoa package? Maybe this would help here.

Yes, this would be a solution.

> So, how can I tell the difference between the two in ObjC code?

You can't, both are public.

> Can't we split up the classes into two headers? 

No, its an auto generated header by Xcode

Reply:
I would appreciate it if we didn't make all the Swift classes public to fix this issue, especially if we plan on converting more code to Swift. So even though moving the SR code to its extra package seems like a lot of work to fix this, it's the right thing to do, IMO. It would also reduce the size of the sentry-cocoa lib.